src/App.tsx

```diff
@@ -1,10 +1,38 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer'; // Import the new Footer component
+import { Todo } from './types/Todo';
+import { ErrorMessage } from './types/ErrorMessage';
+import { filter, FilterType } from './utils/filter';
+import { getTodos } from './api/todos';
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | null>(null);
+  const [shownTodos, setShownTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState(FilterType.ALL);
+
+  useEffect(() => {
+    getTodos()
+      .then(todos => setTodosFromServer(todos))
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  useEffect(
+    () => setShownTodos(filter(todosFromServer, filterType)),
+    [todosFromServer, filterType],
+  );
```

Comment:
we dont need second useEffect and second state

use const instead like const filterdTodos = filter(todosFromServer, filterType)